packages/react-dom/src/client/DOMPropertyOperations.js

```diff
@@ -125,84 +100,59 @@ export function getValueForAttribute(node, name, expected) {
-  const propertyInfo = getPropertyInfo(name);
-
-  if (propertyInfo && shouldSetAttribute(name, value)) {
-    if (shouldIgnoreValue(propertyInfo, value)) {
-      deleteValueForProperty(node, name);
-      return;
-    } else if (propertyInfo.mustUseProperty) {
-      // Contrary to `setAttribute`, object properties are properly
-      // `toString`ed by IE8/9.
-      node[propertyInfo.propertyName] = value;
-    } else {
-      const attributeName = propertyInfo.attributeName;
-      const namespace = propertyInfo.attributeNamespace;
-      // `setAttribute` with objects becomes only `[object]` in IE8/9,
-      // ('' + value) makes it output the correct toString()-value.
-      if (namespace) {
-        node.setAttributeNS(namespace, attributeName, '' + value);
-      } else if (
-        propertyInfo.hasBooleanValue ||
-        (propertyInfo.hasOverloadedBooleanValue && value === true)
-      ) {
-        node.setAttribute(attributeName, '');
+export function setValueForProperty(node, name, value, isCustomComponentTag) {
+  if (shouldSkipAttribute(name, isCustomComponentTag)) {
+    return;
+  }
+  const propertyInfo = isCustomComponentTag ? null : getPropertyInfo(name);
+  if (shouldTreatAttributeValueAsNull(name, value, isCustomComponentTag)) {
+    value = null;
+  }
+  // If the prop is in the special list, treat it as a simple attribute.
+  if (!propertyInfo) {
+    if (isAttributeNameSafe(name)) {
+      const attributeName = name;
+      if (value == null) {
+        node.removeAttribute(attributeName);
       } else {
         node.setAttribute(attributeName, '' + value);
       }
     }
-  } else {
-    setValueForAttribute(
-      node,
-      name,
-      shouldSetAttribute(name, value) ? value : null,
-    );
     return;
   }
-}
-
-export function setValueForAttribute(node, name, value) {
-  if (!isAttributeNameSafe(name)) {
+  const {
+    hasBooleanValue,
+    hasOverloadedBooleanValue,
+    mustUseProperty,
+  } = propertyInfo;
+  if (mustUseProperty) {
+    const {propertyName} = propertyInfo;
+    if (value === null) {
+      node[propertyName] = hasBooleanValue ? false : '';
+    } else {
+      // Contrary to `setAttribute`, object properties are properly
+      // `toString`ed by IE8/9.
+      node[propertyName] = value;
+    }
     return;
   }
-  if (value == null) {
-    node.removeAttribute(name);
+  // The rest are treated as attributes with special cases.
+  const {attributeName, attributeNamespace} = propertyInfo;
+  if (value === null) {
+    node.removeAttribute(attributeName);
   } else {
-    node.setAttribute(name, '' + value);
-  }
-}
-
-/**
- * Deletes an attributes from a node.
- *
- * @param {DOMElement} node
- * @param {string} name
- */
-export function deleteValueForAttribute(node, name) {
-  node.removeAttribute(name);
-}
-
-/**
- * Deletes the value for a property on a node.
- *
- * @param {DOMElement} node
- * @param {string} name
- */
-export function deleteValueForProperty(node, name) {
```

Comment:
Do we care? There's plenty of internal names, even in test files (e.g. `ReactDOMComponent-test`). I agree we need to clean it up by functionality rather than file, but don't think it's necessary to do in the scope of this PR.

Reply:
Using names of internal APIs that exist is different. In this case, it's referrencing a function that no longer exists. If I was looking to contribute and encountered that, I would be confused that the function didn't exist anywhere in the codebase.

It's not critical, but since you remove it feels like it's in scope to fix any lingering references. If we're going to restructure the tests soon anyways, it probably doesn't matter.

Reply:
I guess what I'm referring to is that there are already plenty examples where internal names don't match tests. It used to be important for them to be in sync when unit test tested internals, but they don't anymore. I agree it's confusing, but IMO mismatches in test file names are even more confusing.

Conceptually those tests *are* about deleting. I wouldn't move them into `set*ForProperty` just because those code paths moved there. Instead I would've preferred to rename all tests to human readable descriptions (e.g. "deletion"). But then it's already inconsistent in the current test codebase so I think it's easier to do in one big cleanup later.